go-controller/pkg/ovndbmanager/ovndbmanager_test.go

```diff
@@ -132,3 +613,45 @@ func TestElectionTimer(t *testing.T) {
 func keyForArgs(args ...string) string {
 	return strings.Join(args, "-")
 }
+
+// create file with name, fail on error or if the file already exists
+func createDbFile(t *testing.T, name string, create bool) {
```

Comment:
@andreaskaris best practice is to create a temp dir with t.TempDir() at the beginning of the test in which you can write files. That gets cleaned up for you automatically. So I would do that at the top of `TestResetRaftDB` and then pass that directory to `createDbFile()` and `filepath.Join()` the dbname to that directory.

Reply:
I added this to the top of the Test function:
~~~
func TestResetRaftDB(t *testing.T) {
	// resetRaftDB expects a file named ovnnb or ovnsb in the local directory.
	// Change to a temporary directory for this test and switch back once the test returns.
	tmpDir := t.TempDir()
	currentDir, err := os.Getwd()
	if err != nil {
		t.Fatalf("Cannot get current directory: %s", err.Error())
	}
	err = os.Chdir(tmpDir)
	if err != nil {
		t.Fatalf("Cannot change to temporaryr directory: %s", err.Error())
	}
	t.Logf("Switched to temporary directory: %s", tmpDir)
	defer func() {
		err := os.Chdir(currentDir)
		if err != nil {
			t.Fatalf("Cannot change back to current directory: %s", err.Error())
		}
		t.Logf("Switched back to current directory: %s", currentDir)
	}()
~~~

The beginning of resetRaftDb looks like this:
~~~
// resetRaftDB backs up the db by renaming it and then stops the nb/sb ovsdb process.
// Returns an error if anything goes wrong.
func resetRaftDB(db *dbProperties) error {
        dbFile := filepath.Base(db.dbAlias)
        backupFile := strings.TrimSuffix(dbFile, filepath.Ext(dbFile)) +
                time.Now().UTC().Format("2006-01-02_150405") + "db_bak"
        backupDB := filepath.Join(filepath.Dir(db.dbAlias), backupFile)
        err := os.Rename(db.dbAlias, backupDB)
        if err != nil {
                return fmt.Errorf("failed to back up the db to backupFile: %s", backupFile)
        }
~~~

So I think that I either have to os.Chdir to the temp directory (as in the current code suggestion) or I have to modify the production code further and add a variable which holds the current directory?